libminifi/include/core/ConfigurableComponent.h

```diff
@@ -216,16 +216,20 @@ bool ConfigurableComponent::getProperty(const std::string name, T &value) const{
 
    auto &&it = properties_.find(name);
    if (it != properties_.end()) {
-     Property item = it->second;
-     value = static_cast<T>(item.getValue());
-     if (item.getValue().getValue() != nullptr){
-       logger_->log_debug("Component %s property name %s value %s", name, item.getName(), item.getValue().to_string());
-       return true;
-     }
-     else{
+     const Property& item = it->second;
+     if (item.getValue().getValue() == nullptr) {
+       // empty value
+       if (item.getRequired()) {
+         logger_->log_debug("Component %s required property %s is empty", name, item.getName());
+         throw utils::RequiredPropertyMissingException("Required property is empty: " + item.getName());
```

Comment:
definitely, if I create a processor and mark a property required and it is not there, it is the fault of the processor scheduler and I should not handle its absence (should not catch the exception but let it up to the scheduler), on the other hand if I have an optional property and it is not there, tough luck but I'll have to manage its absence

Reply:
I am not saying that the logic is bad, just that this requires knowledge of the implementation details of `getProperty`. My expectation on calling a function of `bool getProperty(std::string)` is that I get a false returned if there is an error, and I might miss the fact that this throws. It would either be a good idea to show that this throws-if-required-missing in the name of the function or move the whole is-required check up to the scope of the caller.

Reply:
the thing is, that you as a processor implementer should not care if `getProperty` throws or not, returning `false` means: "hey some error occurred that you should handle", throwing an exception means that somebody upstream messed up, e.g. called `onSchedule` without validating if all required properties are present or not checking invalid properties, if I mark a property required or add a validator I do not want to be have to handle if it is missing or invalid